x/auth/ante/unordered.go

```diff
@@ -62,7 +71,27 @@ func (d *UnorderedTxDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, _ bool, ne
 		return ctx, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "unordered tx ttl exceeds %d", d.maxUnOrderedTTL)
 	}
 
-	txHash := sha256.Sum256(ctx.TxBytes())
+	// in order to create a deterministic hash based on the tx, we need to hash the contents of the tx with signature
+	// Get a Buffer from the pool
+	buf := bufPool.Get().(*bytes.Buffer)
+	// Make sure to reset the buffer
+	buf.Reset()
+
+	// Use the buffer
+	for _, msgs := range tx.GetMsgs() {
+		buf.Write([]byte(msgs.String()))
+	}
+
+	sigTx, ok := tx.(sdk.Signature)
+	if !ok {
+		return ctx, errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "tx must implement Signature interface")
+	}
+	buf.Write(sigTx.GetSignature())
+
+	txHash := sha256.Sum256(buf.Bytes())
```

Comment:
is this enough or should i include more information from the transaction 

cc @aaronc @testinginprod

Reply:
Including signatures actually opens up a route to malleability, probably not in secp256k1 but maybe in other algorithms. `msg.String()` should be assumed to be non deterministic. In fact some implementations intentionally introduce non determinism. Basically there isn't anything in the underlying contract of these methods that ensures determinism and non-malleability.